td-agent/yum/binstubs-test.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+set -exu
+
+# Amazon Linux 2 system-release-cpe is:
+# cpe:2.3:o:amazon:amazon_linux:2
+# CentOS 7 system-release-cpe is:
+# cpe:/o:centos:centos:7
+# CentOS 6 system-release-cpe is:
+# cpe:/o:centos:linux:6
+# This means that column glitch exists.
+# So, we should remove before "o" character.
+
+distribution=$(cat /etc/system-release-cpe | awk '{print substr($0, index($1, "o"))}' | cut -d: -f2)
+version=$(cat /etc/system-release-cpe | awk '{print substr($0, index($1, "o"))}' | cut -d: -f4)
+
+ENABLE_UPGRADE_TEST=1
+case ${distribution} in
+  amazon)
+    case ${version} in
+      2)
+        DNF=yum
+        DISTRIBUTION_VERSION=${version}
+        ;;
+    esac
+    ;;
+  centos)
+    case ${version} in
+      7)
+        DNF=yum
+        DISTRIBUTION_VERSION=${version}
+        ;;
+      *)
+        DNF="dnf --enablerepo=powertools"
+        ENABLE_UPGRADE_TEST=0
+        if [ x"${CENTOS_STREAM}" == x"true" ]; then
+            echo "MIGRATE TO CENTOS STREAM"
+            ${DNF} install centos-release-stream -y && \
+                ${DNF} swap centos-{linux,stream}-repos -y && \
+                ${DNF} distro-sync -y
+            DISTRIBUTION_VERSION=${version}-stream
+        else
+            DISTRIBUTION_VERSION=${version}
+        fi
+        ;;
+    esac
+    ;;
+esac
+
+repositories_dir=/fluentd/td-agent/yum/repositories
+ARCH=$(uname -m)
```

Comment:
For consistency, how about obtain architechtures from `rpm --eval "%{_arch}"`?
We retrieve architecture information with dpkg which is package management commant on apt environments.
We should also retrieve it from package management command which is `rpm`.
WDYT?

Reply:
Thanks!
it is better to use that way. I'll fix it

Reply:
FYI: `uname -m` is almost always returning `x86_64`. This command returns docker container's host kernel architecture.